### PR TITLE
feat: add keywords to results

### PR DESCRIPTION
**Pull request by danielkryska**
All collections in all tags should contain keywords. See below for more context.

- add a possibility to configure secondary tags
- add view, downloads, keywords secondary tags generators
- validate secondary tags & filters configurations

### CODE REVIEW

**Comment by danielkryska** [comment]
image
image
image
image

**Comment by danielkryska** [comment]
image

**Comment by danielkryska** [comment]
image

**Review by michal-szostak — APPROVED** [review]

**Review by agpul — CHANGES_REQUESTED** [review]
1. Why I am not able to filter by "tags" in Services and Data sources? I should be able to.
2.  Please remove keywords duplication from the Training view:

image

---

1. fixed
2. removed


image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
When I click on keyword for Services or Data sources in All catalogs tab
image

**Comment by Marcelinna** [comment]
Can we block adding multiple time the same filtr
image

**Review by Marcelinna — APPROVED** [review]

**Review by Marcelinna — APPROVED** [review]

File: ui/apps/ui/src/app/collections/data/data-sources/adapter.data.ts [hunk 1]
```diff
@@ -54,6 +58,10 @@ export const dataSourcesAdapter: IAdapter = {
         filter: 'resource_organisation',
       },
     ],
-    ...parseStatistics(dataSource),
+    secondaryTags: [
+      toDownloadsStatisticsSecondaryTag(dataSource.usage_counts_downloads),
+      toViewsStatisticsSecondaryTag(dataSource.usage_counts_views),
+      toKeywordsSecondaryTag(dataSource.tag_list ?? [], 'tag_list'),
+    ],
```
Comment (michal-szostak) [inline ui/apps/ui/src/app/collections/data/data-sources/adapter.data.ts:65]: I think this should be a separate function (similar to `parseStatistics(...)`

**Review by michal-szostak — APPROVED** [review]

**Pull request merged**